integration-tests/opentelemetry-reactive/src/test/java/io/quarkus/it/opentelemetry/reactive/enduser/AbstractEndUserTest.java

```diff
@@ -0,0 +1,233 @@
+package io.quarkus.it.opentelemetry.reactive.enduser;
+
+import static io.restassured.RestAssured.get;
+import static io.restassured.RestAssured.given;
+import static java.net.HttpURLConnection.HTTP_OK;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.awaitility.Awaitility.await;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.time.Duration;
+import java.util.Map;
+import java.util.Set;
+
+import org.awaitility.Awaitility;
+import org.hamcrest.Description;
+import org.hamcrest.TypeSafeMatcher;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import io.opentelemetry.semconv.SemanticAttributes;
+import io.quarkus.it.opentelemetry.reactive.Utils;
+
+public abstract class AbstractEndUserTest {
+
+    private static final String END_USER_ID_ATTR = SemanticAttributes.ENDUSER_ID.getKey();
+    private static final String END_USER_ROLE_ATTR = SemanticAttributes.ENDUSER_ROLE.getKey();
+    private static final String HTTP_PERM_AUGMENTOR = "HTTP-PERM-AUGMENTOR";
+    protected static final String READER_ROLE = "READER";
+    protected static final String WRITER_ROLE = "WRITER";
+    protected static final String AUGMENTOR_ROLE = "AUGMENTOR";
+
+    /**
+     * This is 'ROLES-ALLOWED-MAPPING-ROLE' role granted to the SecureIdentity by augmentor and
+     * remapped to 'ROLES-ALLOWED-MAPPING-ROLE-HTTP-PERM' role which allows to verify that the
+     * 'quarkus.http.auth.roles-mapping' config-level roles mapping is reflected in the End User attributes.
+     */
+    protected static final String HTTP_PERM_ROLES_ALLOWED_MAPPING_ROLE = "ROLES-ALLOWED-MAPPING-ROLE-HTTP-PERM";
+
+    @BeforeEach
+    @AfterEach
+    public void reset() {
+        given().get("/reset").then().statusCode(HTTP_OK);
+        await().atMost(5, SECONDS).until(() -> Utils.getSpans().isEmpty());
+    }
+
+    @Test
+    public void testAttributesWhenNoAuthorizationInPlace() {
+        assertEndUserAttributes("/no-authorization", isProactiveAuthEnabled(), User.SCOTT, null);
```

Comment:
I think this is still hard to understand... 
This assertion only defines the input params, not the values to expect... Would it be possible to assert a list of expected roles and the user present?